File: app/src/main/java/com/example/chap19example/MainActivity.java
package com.example.chap19example;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;
import android.widget.Toast;

import java.util.Objects;

public class MainActivity extends AppCompatActivity {

    Button toast;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        // If the following function keeps causing java.lang.NullPointerException,
        // make sure you use AppCompat in your theme.
        getSupportActionBar().setDisplayShowHomeEnabled(true);
        getSupportActionBar().setLogo(R.mipmap.steven);
        getSupportActionBar().setDisplayUseLogoEnabled(true);

        setContentView(R.layout.activity_main);

        toast = findViewById(R.id.button);

        toast.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Toast.makeText(getApplicationContext(), R.string.toast, Toast.LENGTH_LONG).show();
            }
        });
    }
}

File: app/src/main/java/com/example/chap9/MainActivity.kt
package com.example.chap9

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.Button
import android.widget.Toast

class MainActivity : AppCompatActivity() {

    private lateinit var toastButton: Button

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        supportActionBar!!.setDisplayShowHomeEnabled(true)
        supportActionBar!!.setLogo(R.mipmap.steven)
        supportActionBar!!.setDisplayUseLogoEnabled(true)

        setContentView(R.layout.activity_main)

        toastButton = findViewById(R.id.button)

        toastButton.setOnClickListener {
            Toast.makeText(applicationContext, R.string.toast, Toast.LENGTH_LONG).show()
        }
    }
}